index_alias_impl.go

```diff
@@ -641,8 +641,15 @@ func preSearchDataSearch(ctx context.Context, req *SearchRequest, indexes ...Ind
 		}
 	}
 
-	// fix up errors
-	if len(indexErrors) > 0 {
+	// in presearch partial results are not allowed as it can lead to
+	// the real search giving incorrect results, and hence the search
+	// result is reset.
+	// discard partial hits if some child index has failed or
+	// if some child alias has returned partial results.
+	if len(indexErrors) > 0 || sr.Status.Failed > 0 {
+		sr = &SearchResult{
+			Status: sr.Status,
+		}
```

Comment:
What part of the SearchResult were re propagating here that causes the issue?

Reply:
sr.Hits
By calling sr.Merge(asr) earlier, the hits were being appended.
so if any partition errored out, partial results was there, with the errors being 
displayed in the search status

Reply:
Ack.